src/rust/engine/hashing/src/lib.rs

```diff
@@ -312,6 +316,126 @@ impl<W: Write> Write for WriterHasher<W> {
   }
 }
 
+impl<AW: ?Sized + AsyncWrite + Unpin> AsyncWrite for WriterHasher<&mut AW> {
+  fn poll_write(
```

Comment:
Personally I prefer having none public function in the bottom of an impl.

Reply:
This is the same order as defined in `AsyncWrite`